datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -285,6 +286,11 @@ impl ParquetFormat {
         self.options.global.coerce_int96 = time_unit;
         self
     }
+
+    pub fn with_cache_metadata(mut self, cache_metadata: bool) -> Self {
```

Comment:
I think this option was removed in https://github.com/apache/datafusion/pull/17062 so it is no longer needed